javav2/example_code/s3/src/main/java/com/example/s3/ListBuckets.java

```diff
@@ -0,0 +1,39 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.example.s3;
+
+// snippet-start:[s3.java2.list.buckets.main]
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.Bucket;
+import software.amazon.awssdk.services.s3.model.ListBucketsResponse;
+import java.util.List;
+
+/**
+ * Before running this Java V2 code example, set up your development
+ * environment, including your credentials.
+ *
+ * For more information, see the following documentation topic:
+ *
+ * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
+ */
+public class ListBuckets {
+    public static void main(String[] args) {
+        Region region = Region.US_EAST_1;
+        S3Client s3 = S3Client.builder()
+            .region(region)
+            .build();
+
+        listAllBuckets(s3);
+
+    }
+    public static void listAllBuckets(S3Client s3) {
+        ListBucketsResponse response = s3.listBuckets();
+        List<Bucket> bucketList = response.buckets();
```

Comment:
Is there a paginator for this? If so, it should be used.

Reply:
This method not exposed by  S3Client.